client/metric_conditions.go

```diff
@@ -44,20 +41,28 @@ type Thresholds struct {
 }
 
 type MetricQueryWithAttributes struct {
-	Name       string      `json:"query-name"`
-	Type       string      `json:"query-type"`
-	Hidden     bool        `json:"hidden"`
-	Display    string      `json:"display-type"`
-	Query      MetricQuery `json:"metric-query"`
-	SpansQuery SpansQuery  `json:"spans-query,omitempty"`
-	TQLQuery   string      `json:"tql-query"`
+	Name           string         `json:"query-name"`
+	Type           string         `json:"query-type"`
+	Hidden         bool           `json:"hidden"`
+	Display        string         `json:"display-type"`
+	Query          MetricQuery    `json:"metric-query"`
+	SpansQuery     SpansQuery     `json:"spans-query,omitempty"`
+	CompositeQuery CompositeQuery `json:"composite-query,omitempty"`
+	TQLQuery       string         `json:"tql-query"`
 }
 
 type MetricQuery struct {
-	Metric             string        `json:"metric"`
-	Filters            []LabelFilter `json:"filters,omitempty"`
-	TimeseriesOperator string        `json:"timeseries-operator"`
-	GroupBy            GroupBy       `json:"group-by,omitempty"`
+	Metric                          string                `json:"metric"`
+	Filters                         []LabelFilter         `json:"filters,omitempty"`
+	TimeseriesOperator              string                `json:"timeseries-operator"`
+	TimeseriesOperatorInputWindowMs *int                  `json:"timeseries-operator-input-window-ms,omitempty"`
```

Comment:
just curious: is the `omitempty` part necessary, or just trying to document the fact that this field is optional?

Reply:
I don't think it is strictly necessary. If the public api get a request that looks like
```
...
"timeseries-operator": "min",
"timeseries-operator-input-window-ms", null,
...
```
I'm pretty sure that would be functionaly equivelant to
```
...
"timeseries-operator": "min",
...
```
I used the omitempty to match the other optional fields in these structs like `Filters`